### PR TITLE
Add overload to handle unknown returned by TS-4.4 for errors be default

### LINKED ISSUE #1115: TypeScript 4.4 breaks logging exceptions from catch
https://github.com/Microsoft/TypeScript-wiki/blob/main/Breaking-Changes.md#using-unknown-in-catch-variables

> Users running with the --strict flag may see new errors around catch variables being unknown due to the new --useUnknownForCatchVariables flag, especially if the existing code assumes only Error values have been caught.

This causes issues with the typing in `pino.error(exception)` for example.

```plain
TS2769: No overload matches this call.
Overload 1 of 2, '(obj: object, msg?: string | undefined, ...args: any[]): void', gave the following error.
Argument of type 'unknown' is not assignable to parameter of type 'object'.
Overload 2 of 2, '(msg: string, ...args: any[]): void', gave the following error.<br/>Argument of type 'unknown' is not assignable to parameter of type 'string'.
```

Of course it works, because in this particular scenario the `error` is an actual thrown `Error`. But this change from TypeScript is because you can in theory throw anything. :shrug:

### PR DESCRIPTION
**Pull request by kibertoad**
fixes #1115

### CODE REVIEW

**Comment by kibertoad** [comment]
@climba03003 Could you please take a look?

**Review by mcollina — APPROVED** [review]
lgtm

File: package.json [hunk 1]
```diff
@@ -94,7 +94,7 @@
     "through2": "^4.0.0",
     "ts-node": "^10.2.1",
     "tsd": "^0.17.0",
-    "typescript": "^4.4.2",
+    "typescript": "4.4.2",
```
Comment (mcollina) [inline package.json:97]: Can you please restore the caret here?

**Review by mcollina — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (kibertoad) [reply package.json]: Done. You prefer random early failures when TS updates to explicit updates? There are benefits to both approaches, I still don't know which one makes more sense to me personally.

Comment (mcollina) [reply package.json]: I just think TS is too much maintenance on dependabot/PRs to manage in that way. If our types start breaking suddenly we'll address it, however that has not been a problem in Fastify so the approach seem to work ok.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**